cpp/tests/utilities/column_utilities.cu

```diff
@@ -118,9 +118,12 @@ std::unique_ptr<column> generate_child_row_indices(lists_column_view const& c,
     0,
     [row_indices = row_indices.begin<size_type>(),
      validity    = c.null_mask(),
+     offsets     = c.offsets().begin<offset_type>(),
      offset      = c.offset()] __device__(int index) {
       auto const true_index = row_indices[index] + offset;
-      return !validity || cudf::bit_is_set(validity, true_index) ? 1 : 0;
+      return !validity || cudf::bit_is_set(validity, true_index)
+               ? (offsets[true_index + 1] - offsets[true_index]) != 0
+               : 0;
```

Comment:
I'm not sure I follow the logic here.  This change would seem to add nulls anywhere there is an empty row, which is not correct.